src/pty.c

```diff
@@ -0,0 +1,86 @@
+#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
+#include <errno.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <signal.h>
+#include <pty.h>
+#include <re.h>
+#include "test.h"
+
+
+#define DEBUG_MODULE "test_pty"
+#define DEBUG_LEVEL 5
+#include <re_dbg.h>
+
+
+struct pty_test {
+	int pty;
+	pid_t pid;
+	int err;
+};
+
+
+static void abort_test(struct pty_test *test, int err)
+{
+	test->err = err;
+	re_cancel();
+}
+
+
+static void pty_hdlr(int flags, void *arg)
+{
+	struct pty_test *test = arg;
+	(void) flags;
+	abort_test(test, 0);
```

Comment:
perhaps the function should check `flags` and abort the test if `FD_EXCEPT` flag is set ?

when running the test, I am getting many `FD_READ` events, and in some cases an `FD_EXCEPT`
at the end.

Reply:
I'm not expecting any input but I agree that this would probably be cleaner.